### PR TITLE
CI: Push docker tags for various forms

### PR DESCRIPTION
**Pull request by oliv3r**
I have no idea if this works, so someone that understands docker's github actions needs to check this. I feel like there's things missing (names mostly from the tags) but it follows the examples/templates from docker itself [build-push-action][0] and [metadata-action][1].

[0]: https://github.com/docker/build-push-action
[1]: https://github.com/docker/metadata-action

### CODE REVIEW

File: .github/workflows/docker.yml [hunk 1]
```diff
@@ -1,8 +1,10 @@
 name: Docker
 on:
   push:
+    branches:
```
Comment (tdewolff) [inline .github/workflows/docker.yml:4]: I want to only execute this GitHub action when releasing a new version, not on every push to the master branch, so this change would have to be reversed

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -1,8 +1,10 @@
...
+      - 'master'
     tags:
-      - 'v*'
+      - 'v*.*.*'
```
Comment (tdewolff) [inline .github/workflows/docker.yml:7]: Not sure why this change, and not sure if it works actually. The usual way is to use `v*` which would match `v2.11.11` or whatever tag that starts with a `v`.

File: .github/workflows/docker.yml [hunk 3]
```diff
@@ -15,6 +17,14 @@ jobs:
         password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
     - name: Set up Docker Buildx
       uses: docker/setup-buildx-action@v1
+    - name: Generate container metadata
+      id: meta
+      uses: docker/metadata=action@v4
+      with:
```
Comment (tdewolff) [inline .github/workflows/docker.yml:23]: I think we should follow the basic example of pushing the tags and not explicitly follow semver (the GitHub tags are already semver). That means we should get rid of this `tags` definition. The tag is automatically used from the GitHub tag.

Additionally, it looks like we need to include `images: tdewolff/minify` here.

**Review by tdewolff — COMMENTED** [review]
Thanks for the PR, added some small comments ;-)

File: .github/workflows/docker.yml (hunk 1 above)
Comment (oliv3r) [reply .github/workflows/docker.yml:4]: Right, I just followed the template from docker; I have no idea what is needed and why it would work; You could decide to push a `:dev` or `:master` is what the intend here is; but sure, we can drop it :)

File: .github/workflows/docker.yml (hunk 2 above)
Comment (oliv3r) [reply .github/workflows/docker.yml:7]: yeah also here, from the template ;)

File: .github/workflows/docker.yml (hunk 3 above)
Comment (oliv3r) [reply .github/workflows/docker.yml:23]: I have no idea; could be; as for the tags, I think what this action also tries to do, is do  a 'on push of `v3.1.2`, also push `v3.1` and `v3`, which is why you'd define them.

**Comment by tdewolff** [comment]
I've added the changes, should work from the next version onwards!

**Comment by oliv3r** [comment]
Thanks @tdewolff!!
